default/python/universe_generation/starsystems.py

```diff
@@ -101,20 +111,16 @@ def generate_systems(pos_list, gsd):
             if planets.generate_a_planet(system, star_type, orbit, gsd.planet_density, gsd.shape):
                 at_least_one_planet = True
 
-        if star_type == fo.starType.blackHole:
+        if at_least_one_planet or can_have_no_planets(star_type):
             continue
 
-        ii_safety = 1000
-        while not at_least_one_planet and ii_safety > 0:
-            ii_safety -= 1
-            for orbit in orbits:
-                if planets.generate_a_planet(system, star_type, orbit, gsd.planet_density, gsd.shape):
-                    at_least_one_planet = True
-                    break
-
-        if not at_least_one_planet:
+        for _unused_recursion_guard, orbit in product(range(1000), orbits):
```

Comment:
I've refactored this line so it should be a non-issue.

I think of the inner loop as a recursive function that keeps getting called until it generates a planet or it hits the recursion limit.